realm-annotations/src/main/java/io/realm/annotations/RealmField.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2018 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.annotations;
+
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation used on fields in Realm model classes. It describes metadata about the field.
+ */
+@Retention(RetentionPolicy.RUNTIME)
```

Comment:
Because I asked! 😄 

The other annotations like `@RealmModule` and `@Index` and `@Required` were changed to `RUNTIME` so that they can be accessed via reflection, this way you can read this data without having to write your own annotation processor.

Like here https://github.com/realm/realm-java/issues/5344